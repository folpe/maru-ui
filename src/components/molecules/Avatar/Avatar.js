import React from 'react'

import { node, oneOf, string } from 'prop-types'

import './avatar.scss'
import Badge from '../../atoms/Badge'
import { getInitials } from '../../../helpers/common'

/**
 * Represents an avatar.
 * @component
 * @param {image} image - The image of avatar .
 * @param {name} name - The name of the user 'Firstname' || 'Pseudo'.
 * @param {lastname} lastname - The lastname of the user 'Lastname'.
 * @param {size} size - The size of avatar ['small', 'medium', 'large']
 * @param {form} form - The form of avatar ['rounded', 'circle'].
 * @param {color} color - The color of avatar ['primary', 'secondary'].
 */
const Avatar = ({ image, name, lastname, size, form, color, ...props }) => {
  return (
    <Badge className='avatar' size={size} color={color} form={form}>
      {image || <span>{getInitials({ name, lastname })}</span>}
    </Badge>
  )
}

Avatar.propTypes = {
  size: oneOf(['small', 'medium', 'large']),
  name: string,
  lastname: string,
  form: oneOf(['rounded', 'circle']),
  color: oneOf(['primary', 'secondary']),
  image: node,
}

export default Avatar
