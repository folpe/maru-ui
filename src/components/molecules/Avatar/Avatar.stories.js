import React from 'react'

import Avatar from './Avatar'
import imageFile from '../../../../static/avatar.jpg'

const image = {
  src: imageFile,
  alt: 'my image',
}

const img = <img src={image.src} alt={image.alt} />

export default {
  title: 'Molecules',
  component: Avatar,
}

const Template = (args) => <Avatar {...args} />

export const AvatarBasic = Template.bind({})
AvatarBasic.args = {
  name: 'Shiba',
  lastname: 'Maru',
}

export const AvatarImage = Template.bind({})
AvatarImage.args = {
  image: img,
}
