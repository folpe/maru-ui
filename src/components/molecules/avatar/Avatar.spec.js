import React from 'react'
import { render } from '@testing-library/react'

import Avatar from './Avatar'

describe('<Avatar />', () => {
  it('should otherwise match snapshot', () => {
    const { container } = render(
      <Avatar size='small' name='Shiba' lastname='Maru' />
    )
    expect(container).toMatchSnapshot()
  })

  it('should display an image if props exist', () => {})
  it('should display initials if no image passed', () => {})
})
