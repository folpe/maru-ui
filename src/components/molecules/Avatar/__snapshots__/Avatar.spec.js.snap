// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> should otherwise match snapshot 1`] = `
<div>
  <div
    class="badge circle primary small avatar"
  >
    <span>
      SM
    </span>
  </div>
</div>
`;
